### PR TITLE
Try to build an ARM64 image

### PR DESCRIPTION
**Pull request by gillg**
**Description:** 
To be able installing otel-collector-contrib on ARM64 oriented nodes (like Amazon Graviton backed CPU) build an image for x64 and another for arm64.

**Notes for mainteners:**

I'm not sure if buildx is enabled on your pipeline platform...
Once done both images have to be pushed on remote repositories.

### CODE REVIEW

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
Unfortunately we broke the main branch, need to do more work to rebase to the correct main

**Comment by gillg** [comment]
@bogdandrutu my branch has been completly rebased on the last main branch.

File: Makefile [hunk 1]
```diff
@@ -169,9 +169,10 @@ run:
 .PHONY: docker-component # Not intended to be used directly
 docker-component: check-component
 	GOOS=linux GOARCH=amd64 $(MAKE) $(COMPONENT)
-	cp ./bin/$(COMPONENT)_linux_amd64 ./cmd/$(COMPONENT)/$(COMPONENT)
-	docker build -t $(COMPONENT) ./cmd/$(COMPONENT)/
-	rm ./cmd/$(COMPONENT)/$(COMPONENT)
+	GOOS=linux GOARCH=arm64 $(MAKE) $(COMPONENT)
+	cp ./bin/$(COMPONENT)_* ./cmd/$(COMPONENT)/
```
Comment (gillg) [inline Makefile:173]: We copy all binaries to handle more future platform if needed.

File: cmd/otelcontribcol/Dockerfile [hunk 2]
```diff
@@ -5,9 +5,11 @@ FROM scratch
 
 ARG USER_UID=10001
 USER ${USER_UID}
+# Args automaticaly filled by buildkit
+ARG TARGETOS TARGETARCH
```
Comment (gillg) [inline cmd/otelcontribcol/Dockerfile:9]: Changing `buildx build --platform linux/amd64,linux/arm64` will automaticaly change these args, so we can add new platforms tomorrow if we build binaries for other platorms, just changing platforms targeted in the build command.

File: Makefile [hunk 3]
```diff
@@ -169,9 +169,10 @@ run:
 .PHONY: docker-component # Not intended to be used directly
 docker-component: check-component
 	GOOS=linux GOARCH=amd64 $(MAKE) $(COMPONENT)
```
Comment (gillg) [inline Makefile:171]: Can't we reuse binaries from `otelcontribcol-all-sys` ? It avoids to declare manualy each platform to build.

**Comment by gillg** [comment]
Hello ! Can I help with something here or improve something ? Does someone has plans to test this change in pipeline ?

**Pull request closed without merging**